src/syntax_tree.jl

```diff
@@ -141,12 +141,13 @@ function _show_syntax_node(io, current_filename, node, indent)
     end
 end
 
-function _show_syntax_node_sexpr(io, node)
+function _show_syntax_node_sexpr(io, node::SyntaxNode)
     if !haschildren(node)
         if is_error(node)
             print(io, "(", untokenize(head(node)), ")")
         else
-            print(io, node.val isa Symbol ? string(node.val) : repr(node.val))
+            val = node.val
+            print(io, val isa Symbol ? string(val) : repr(val))
```

Comment:
I'm curious - is this stylistic or does it do something more?

Ah wait I think I see - this allows inference to prove that `val::Symbol` inside the first branch.

Reply:
Yeah currently inference can propagate conditional type information about local variable but not about memory owned by local variable. So we need to extract the field value into a local variable to get that extra type propagation.